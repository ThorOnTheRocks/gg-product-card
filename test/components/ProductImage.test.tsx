import React from 'react';
import ProductCard, { ProductImage } from '../../src/components';
import renderer from 'react-test-renderer';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  // test('it should render the component correctly with a personalized title', () => {
  //   const wrapper = renderer.create(
  //     <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
  //   );
  //   expect(wrapper.toJSON()).toMatchInlineSnapshot(`
  //     <div
  //       className="productCard undefined"
  //     >
  //       <img
  //         alt="coffee mug"
  //         className="productImg undefined"
  //         src="./coffee-mug.png"
  //       />
  //     </div>
  //   `);

  test('it should render the component correctly with a personalized image', () => {
    const wrapper = renderer.create(<ProductImage image="https://hello.com" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('it should render the component correctly with a personalized title', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchInlineSnapshot(`
      <div
        className="productCard undefined"
      >
        <img
          alt="coffee mug"
          className="productImg undefined"
          src="./coffee-mug.png"
        />
      </div>
    `);
  });
});
