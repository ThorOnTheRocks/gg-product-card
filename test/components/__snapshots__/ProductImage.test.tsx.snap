// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage it should render the component correctly with a personalized image 1`] = `
<img
  alt="coffee mug"
  className="productImg undefined"
  src="https://hello.com"
/>
`;
